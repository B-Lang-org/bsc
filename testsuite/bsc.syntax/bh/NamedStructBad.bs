package NamedStructBad where

data Baz = Quux { foo :: Bool; bar :: String }
         | FooBar Bool String

testFooOK :: Baz -> Bool
testFooOK (Quux q) = q.foo
testFooOK (FooBar foo _) = foo

testBarBad :: Baz -> String
testBarBad (Quux _ bar) = bar
testBarBad (FooBar fb) = fb._2

