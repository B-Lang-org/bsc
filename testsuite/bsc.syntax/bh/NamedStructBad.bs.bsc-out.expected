checking package dependencies
compiling NamedStructBad.bs
Error: "NamedStructBad.bs", line 11, column 12: (T0142)
  Pattern for constructor `Quux' of type NamedStructBad.Baz requires 1
  arguments, found 2.
